test/integration/instances_test.go

```diff
@@ -589,6 +589,206 @@ func TestInstance_withPG(t *testing.T) {
+			t.Errorf("Error deleting test Instance: %s", terr)
+		}
+	}
+
+	return instance, teardown, nil
+}
+
+func TestInstance_CreateWithOwnedNonAssignedReservedIP(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithOwnedNonAssignedReservedIP")
+	defer teardown()
+
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Errorf("Unexpected error with owned non-assigned reserved IP: %v", err)
+	} else {
+		defer instanceTeardown()
+	}
+}
+
+func TestInstance_CreateWithAlreadyAssignedReservedIP(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithAlreadyAssignedReservedIP")
+	defer teardown()
+
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	// First, create an instance with the reserved IP
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Fatalf("Failed to create initial instance: %v", err)
+	}
+	defer instanceTeardown()
+
+	// Now try to create another instance with the same IP
+	_, _, err = createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err == nil {
+		t.Errorf("Expected error with already assigned reserved IP, but got none")
+	}
+}
+
+func TestInstance_CreateWithNonReservedAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithNonReservedAddress")
+	defer teardown()
+
+	_, _, err := createInstanceWithReservedIP(t, client, "192.0.2.1")
+	if err == nil {
+		t.Errorf("Expected error with non-reserved address, but got none")
+	}
+}
+
+func TestInstance_CreateWithNonOwnedReservedAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithNonOwnedReservedAddress")
+	defer teardown()
+
+	_, _, err := createInstanceWithReservedIP(t, client, "198.51.100.1")
+	if err == nil {
+		t.Errorf("Expected error with non-owned reserved address, but got none")
+	}
+}
+
+func TestInstance_CreateWithEmptyIPAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithEmptyIPAddress")
+	defer teardown()
+
+	_, _, err := createInstanceWithReservedIP(t, client, "")
+	if err == nil {
+		t.Errorf("Expected error with empty IP address, but got none")
```

Comment:
No tear-down for this creation

Reply:
Added teardown function to handle accidental instance creation

Reply:
Same as above move it right after createInstanceWithReservedIP  and before the error checks. Also, use defer